src/dict.js

```diff
@@ -16,9 +16,14 @@ Sk.builtin.dict = function dict (L) {
     }
 
     this.size = 0;
+    // entries will preserve insertion order of the key value pairs
+    this.entries = {};
```

Comment:
I would recommend `this.entries = { __proto__: null }` to create an object with a null prototype -- this will avoid issues with accesses to `Object.prototype`, and in V8 it hints that this object wants to be in dictionary mode.

Reply:
Yes this is probably a good idea. I think there are only 3 values that could clash
`_proto__, _defineGetter__, _defineSetter__` 
but stranger things can happen!!!
will use `Object.create(null)`